pyiron_contrib/atomistic/interactive/montecarlo.py

```diff
@@ -0,0 +1,359 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+import numpy as np
+import random
+import scipy.constants as sci_const
+from pyiron_base import GenericParameters
+from pyiron.lammps.interactive import LammpsInteractive
+from pyiron.atomistics.job.interactivewrapper import InteractiveWrapper
+from pyiron.atomistics.structure.atoms import Atoms
+
+__author__ = "Jan Janssen"
+__copyright__ = "Copyright 2020, Max-Planck-Institut für Eisenforschung GmbH - " \
+                "Computational Materials Design (CM) Department"
+__version__ = "1.0"
+__maintainer__ = "Jan Janssen"
+__email__ = "janssen@mpie.de"
+__status__ = "development"
+__date__ = "Jul 25, 2018"
+
+
+class MonteCarloMaster(InteractiveWrapper):
+    """
+    A reference implementation for Lattice Monte Carlo Simulation. To understand Monte Carlo Simulation we highly
+    recommend implementing your own Monte Carlo Method, still this is a reference implementation to compare your
+    results. On a predefined lattice, the atoms are inserted as interstitials and the energy is calculated using
+    a quantum engine, which can be set as reference job.
+
+    Args:
+        project (ProjectHDFio): ProjectHDFio instance which points to the HDF5 file the job is stored in
+        job_name (str): name of the job, which has to be unique within the project
+
+    Attributes:
+        input (pyiron_base.GenericParameters instance): Instance which handles the input
+        structure_lattice (pyiron.atomistics.structure.atoms.Atoms): Crystal Lattice
+        structure_sublattice (pyiron.atomistics.structure.atoms.Atoms): Crystal Lattice of interstitial positions
+        ref_job (pyiron.atomistics.job.atomistic.AtomisticGenericJob): Quantum Engine used to evaluate structures
```

Comment:
As with `Mixer`, the type is `AtomisticGenericJob`, so references to "quantum" should be purged in favour of something more generic.